lite/tests/classification/test_rocauc.py

```diff
@@ -0,0 +1,254 @@
+    assert rocauc[1] == 0.625  # (animal, dog)
+    assert rocauc[2] == 1.0  # (animal, cat)
+    assert rocauc[3] == 0.75  # (color, white)
+    assert rocauc[4] == 0.875  # (color, red)
+    assert rocauc[5] == 0.0  # (color, blue)
+    assert rocauc[6] == 0.09999999999999998  # (color, black)
+
+    # test mROCAUC
+    assert mean_rocauc.shape == (np.unique(label_metadata[:, 2]).size,)
+    assert (
+        mean_rocauc == np.array([0.8009259259259259, 0.43125])  # animal, color
+    ).all()
+
+
+def test_rocauc_with_example(
+    classifications_two_categeories: list[Classification],
+):
+
+    loader = DataLoader()
+    loader.add_data(classifications_two_categeories)
+    evaluator = loader.finalize()
+
+    metrics = evaluator.evaluate()
+
+    # test ROCAUC
+    actual_metrics = [m.to_dict() for m in metrics[MetricType.ROCAUC]]
+    expected_metrics = [
+        {
+            "type": "ROCAUC",
+            "value": 0.7777777777777778,
+            "parameters": {
+                "label": {"key": "animal", "value": "bird"},
+            },
+        },
+        {
+            "type": "ROCAUC",
+            "value": 0.625,
+            "parameters": {
+                "label": {"key": "animal", "value": "dog"},
+            },
+        },
+        {
+            "type": "ROCAUC",
+            "value": 1.0,
+            "parameters": {
+                "label": {"key": "animal", "value": "cat"},
+            },
+        },
+        {
+            "type": "ROCAUC",
+            "value": 0.75,
+            "parameters": {
+                "label": {"key": "color", "value": "white"},
+            },
+        },
+        {
+            "type": "ROCAUC",
+            "value": 0.875,
+            "parameters": {
+                "label": {"key": "color", "value": "red"},
+            },
+        },
+        {
+            "type": "ROCAUC",
+            "value": 0.0,
+            "parameters": {
+                "label": {"key": "color", "value": "blue"},
+            },
+        },
+        {
+            "type": "ROCAUC",
+            "value": 0.09999999999999998,
+            "parameters": {
+                "label": {"key": "color", "value": "black"},
+            },
+        },
+    ]
+    for m in actual_metrics:
+        assert m in expected_metrics
+    for m in expected_metrics:
+        assert m in actual_metrics
+
+    # test mROCAUC
+    actual_metrics = [m.to_dict() for m in metrics[MetricType.mROCAUC]]
+    expected_metrics = [
+        {
+            "type": "mROCAUC",
+            "value": 0.8009259259259259,
+            "parameters": {
+                "label_key": "animal",
+            },
+        },
+        {
+            "type": "mROCAUC",
+            "value": 0.43125,
+            "parameters": {
+                "label_key": "color",
+            },
+        },
+    ]
+    for m in actual_metrics:
+        assert m in expected_metrics
+    for m in expected_metrics:
+        assert m in actual_metrics
```

Comment:
I found an unexpected result when writing this test:

```python

def test_nonoverlapping_labels():
    animal_gts = ["cat"]
    animal_pds = [
        {"bird": 0.6, "dog": 0.2, "cat": 0.2},
    ]

    classifications = [
        Classification(
            uid=f"uid{idx}",
            groundtruths=[("animal", gt)],
            predictions=[("animal", key) for key, _ in pd.items()],
            scores=[score for score in pd.values()],
        )
        for idx, (gt, pd) in enumerate(zip(animal_gts, animal_pds))
    ]

    loader = DataLoader()
    loader.add_data(classifications)
    evaluator = loader.finalize()

    metrics = evaluator.evaluate(score_thresholds=[0.05, 0.5, 0.95])
    metrics_to_test = [
        MetricType.Accuracy,
        MetricType.Precision,
        MetricType.ROCAUC,
    ]

    actual_metrics = [
        m.to_dict() for key in metrics_to_test for m in metrics[key]
    ]
    expected_metrics = [
        {
            "type": "Accuracy",
            "value": [0.0, 0.0, 0.0],
            "parameters": {
                "score_thresholds": [0.05, 0.5, 0.95],
                "hardmax": True,
                "label": {"key": "animal", "value": "cat"},
            },
        },
        {
            "type": "Precision",
            "value": [0.0, 0.0, 0.0],
            "parameters": {
                "score_thresholds": [0.05, 0.5, 0.95],
                "hardmax": True,
                "label": {"key": "animal", "value": "cat"},
            },
        },
        {
            "type": "ROCAUC",
            "value": 0,
            "parameters": {"label": {"key": "animal", "value": "cat"}},
        },
        # TODO should we return ROCAUC == 0 for "bird" and "dog" here if we don't return any Precision / Accuracy metrics for these two labels?
        {
            "type": "ROCAUC",
            "value": 0,
            "parameters": {"label": {"key": "animal", "value": "bird"}},
        },
        {
            "type": "ROCAUC",
            "value": 0,
            "parameters": {"label": {"key": "animal", "value": "dog"}},
        },
    ]
    for m in actual_metrics:
        assert m in expected_metrics
    for m in expected_metrics:
        assert m in actual_metrics
```

should we return ROCAUC metrics for `bird` and `dog` in this example when we don't return Accuracy or Precision metrics for these two labels? seems like we shouldn't return any metrics for "bird" or "dog" in this case

Reply:
Adjusted the evaluate call to filter out ROCAUC's for labels that are a part of `ignored_prediction_labels`

Reply:
This is what Accuracy, F1, Recall and Precision were already doing